src/mouse_events.cpp

```diff
@@ -785,17 +787,23 @@ void mouse_handler::select_or_action(bool browse)
 	wb::future_map_if_active planned_unit_map;
 
 	if(game_lua_kernel* lk = pc_.gamestate().lua_kernel_.get()) {
-		lk->select_hex_callback(last_hex_);
+		lk->select_hex_callback(clicked_hex);
+		// The callback will typically modify clicked_hex, so make sure it's still valid
+		if(!pc_.get_map().on_board(clicked_hex)) return;
```

Comment:
Please put the `return` on its own line. In the middle of a function like this, inside a conditional block, isn't an obvious place to look for a check-or-return statement.

Reply:
What a pain…